quarkus cli fail hard on bad settings.xml

### Describe the bug

make an error in `~/.m2/settings.xml` like put a start element but no end element.

now quarkus create app gives you:

```
quarkus create app app --stream=2.13 --jbang
[ERROR] ❗  Unable to create project: Failed to initialize the Maven artifact resolver
```

No extra info.

Fix the typos in settings.xml and it works.

You can still make invalid settings.xml but with balanced tags  - then you get an actual exception printed as warning.l

looks like we are not catching and reporting properly the maven resolver creation in these cases.

### Expected behavior

_No response_

### Actual behavior

report failure to parse the settings.xml file.

### How to Reproduce?

_No response_

### Output of `uname -a` or `ver`

_No response_

### Output of `java -version`

_No response_

### GraalVM version (if different from Java)

_No response_

### Quarkus version or git rev

_No response_

### Build tool (ie. output of `mvnw --version` or `gradlew --version`)

_No response_

### Additional information

_No response_

--- a/independent-projects/tools/devtools-common/src/main/java/io/quarkus/devtools/project/QuarkusProjectHelper.java
+++ b/independent-projects/tools/devtools-common/src/main/java/io/quarkus/devtools/project/QuarkusProjectHelper.java
@@ -237,7 +237,18 @@ public class QuarkusProjectHelper {
                         .setWorkspaceDiscovery(false)
                         .build();
             } catch (BootstrapMavenException e) {
-                throw new IllegalStateException("Failed to initialize the Maven artifact resolver", e);
+                StringBuilder messages = new StringBuilder("Failed to initialize the Maven artifact resolver");
+                Throwable current = e;
+                while (null != current) {
+                    if (null != current.getMessage() && !current.getMessage().isBlank()) {
+                        messages
+                                .append("\n")
+                                .append(current.getMessage());
+                    }
+
+                    current = current.getCause();
+                }
+                throw new IllegalStateException(messages.toString().trim(), e);
             }
         }
         return artifactResolver;

